crates/rspack_plugin_javascript/src/plugin/mod.rs

```diff
@@ -86,9 +86,17 @@ impl JsPlugin {
             execOptions.factory.call(module.exports, module, module.exports, execOptions.require);
             "#,
       ),
-      false => RawSource::from(
-        "__webpack_modules__[moduleId](module, module.exports, __webpack_require__);\n",
-      ),
+      false => {
+        if runtime_requirements.contains(RuntimeGlobals::THIS_AS_EXPORTS) {
```

Comment:
Runtime should be changed by this_as_exports, otherwise this.xxx will not take effect.